src/auth/Readme.md

```diff
@@ -0,0 +1,266 @@
+
+#### **`useSaleorAuthClient({ saleorApiUrl, storage, onAuthRefresh }: UseSaleorAuthClientProps) => SaleorAuthClientProps`**
+
+On top of taking part in react component lifecycle, because the client sets event listeners on the window, we need to remove those once we're done. Hence we're proving an `useSaleorAuthClient` hook that does exactly that. It takes up the same props as `SaleorAuthClient` and returns the instance of SaleorAuthClient along with isAuthenticating prop.
+
+```javascript
+const { saleorAuthClient, isAuthenticating } = useSaleorAuthClient({
+  saleorApiUrl: "https://master.staging.saleor.cloud",
+  storage: window.localStorage,
+});
+```
+
+## How do I use the auth without prop drilling?
+
+#### **`SaleorAuthProvider({ saleorAuthClient, isAuthenticating, children }: PropsWithChildren<UseSaleorAuthClient>) => JSX.Element`**
+
+The hook is supposed to be used only in the root of your project. If you'd like to access the client methods or isAuthenticating prop down the tree without extensive prop drilling, you can use `SaleorAuthProvider` passing it down the return value of useSaleorAuthClient hook.
+
+```jsx
+const saleorAuthClientData = useSaleorAuthClient(authClientProps)
+
+<SaleorAuthProvider {...saleorAuthClientData}>
+    {...}
+</SaleorAuthProvider>
+```
+
+---
+
+#### **`useSaleorAuthContext() => SaleorAuthContextConsumerProps`**
+
+The provider also equips you with `useSaleorAuthContext` hook
+
+```javascript
+const { isAuthenticating, signIn, signOut, checkoutSignOut } = useSaleorAuthContext();
+```
+
+## How do I tell my graphql client to refresh queries on signIn / signOut?
+
+#### **`useAuthChange({ onSignedIn, onSignedOut }: UseAuthChangeProps) => void`**
+
+Once the successful sign in or sign out happens, as well as failed authentication (for e.g when refresh token is expired and there's no way to obtain a new access token from the api) you'll probably want to trigger some actions. Hence we're provided a `useAuthChange` hook that'll listen to storage changes and trigger callbacks.
+
+```javascript
+  useAuthChange({
+    onSignedOut: () => { /* client invalidate some queries */},
+    onSignedIn: () => { /* client invalidate some queries  */}
+    },
+  });
+```
+
+## How do I sign in?
+
+#### **`SaleorAuthClient.signIn: ({ email, password }: TokenCreateVariables) => Promise<TokenCreateResponse>`**
+
+SaleorAuthClient returns a `signIn` method. You can also access it via the **useSaleorAuthContext** hook.
+
+```javascript
+const { signIn } = useSaleorAuthContext();
+
+const response = await signIn({
+  email: "example@mail.com",
+  passowrd: "password",
+});
+```
+
+## How do I sign out?
+
+#### **`SaleorAuthClient.signOut: () => void`**
+
+This method will remove access and refresh tokens from SaleorAuthClient state and storage. It'll also trigger sign out event, which in turn will trigger [onSignOut method of useAuthChange hook](#how-do-i-tell-my-graphql-client-to-refresh-queries-on-signin--signout)
```

Comment:
```suggestion
This method will remove access and refresh tokens from the `SaleorAuthClient` state and storage. It will also call the sign-out event, which will trigger [`onSignOut` method of the `useAuthChange` hook](#how-do-i-tell-my-graphql-client-to-refresh-queries-on-signin--signout).
```